tools/testsys-config/src/lib.rs

```diff
@@ -0,0 +1,416 @@
+    pub fn compute_lock_path<P>(test_config_path: P) -> Result<PathBuf>
+    where
+        P: AsRef<Path>,
+    {
+        Ok(test_config_path
+            .as_ref()
+            .parent()
+            .context(error::ParentSnafu {
+                path: test_config_path.as_ref(),
+            })?
+            .join("Test.lock"))
+    }
+
+    /// Create a single config for the `variant` and `arch` from this config. `starting_config` is
+    /// used as the config with the highest precedence. If no starting config is available,
+    /// `Default::default()` can be used.
+    pub fn reduced_config<S1>(
+        &self,
+        variant: &Variant,
+        arch: S1,
+        starting_config: GenericVariantConfig,
+    ) -> GenericVariantConfig
+    where
+        S1: Into<String>,
+    {
+        let arch = arch.into();
+        config_keys(variant)
+            .into_iter()
+            .filter_map(|key| self.configs.get(&key).cloned())
+            .flat_map(|config| vec![config.for_arch(&arch), config.config])
+            .fold(starting_config, GenericVariantConfig::merge)
+    }
+}
+
+/// High level configurations for a test
+#[derive(Debug, Default, Deserialize, Serialize, PartialEq, Eq, Clone)]
+#[serde(deny_unknown_fields, rename_all = "kebab-case")]
+pub struct Test {
+    /// The name of the repo in `Infra.toml` that should be used for testing
+    pub repo: Option<String>,
+
+    #[serde(flatten)]
+    /// The URI of TestSys images
+    pub testsys_images: TestsysImages,
+
+    /// A registry containing all TestSys images
+    pub testsys_image_registry: Option<String>,
+}
+
+#[derive(Debug, Default)]
+pub struct AwsK8sVariantConfig {
+    /// The names of all clusters this variant should be tested over. This is particularly useful
+    /// for testing Bottlerocket on ipv4 and ipv6 clusters.
+    pub cluster_names: Vec<String>,
+    /// The instance type that instances should be launched with
+    pub instance_type: Option<String>,
+    /// The secrets needed by the agents
+    pub secrets: BTreeMap<String, SecretName>,
+    /// The role that should be assumed for this particular variant
+    pub assume_role: Option<String>,
+    /// The sonobuoy conformance image that should be used for this variant
+    pub kube_conformance_image: Option<String>,
+    /// The e2e repo containing sonobuoy images
+    pub e2e_repo_registry: Option<String>,
+}
+
+#[derive(Debug, Default)]
+pub struct AwsEcsVariantConfig {
+    /// The names of all clusters this variant should be tested over
+    pub cluster_names: Vec<String>,
+    /// The instance type that instances should be launched with
+    pub instance_type: Option<String>,
+    /// The secrets needed by the agents
+    pub secrets: BTreeMap<String, SecretName>,
+    /// The role that should be assumed for this particular variant
+    pub assume_role: Option<String>,
+}
+
+/// Create a vec if relevant keys for this variant ordered from most specific to least specific.
```

Comment:
Nice, this is very lightweight for the power it offers.

```suggestion
/// Create a vec of relevant keys for this variant ordered from most specific to least specific.
```